### ISSUE TITLE
Partial reroute of protected flow after switch mode change may lead to corrupted flow

### ISSUE THREAD

**Issue by rtretyak** [telstra/open-kilda#3341]
0. All switches are in single-table mode
1. create protected flow
2. Change switch table mode on the src switch to multitable
3. Update link costs so that system will only want to reroute main path, but leave protected path intact
4. Call reroute for the flow (check that only main path was rerouted). You can see on src switch that new main path rules are in multitable mode, but protected path rule is still in table 0
5. Call delete for the flow
**Expected:** Flow is removed.
**Actual:** Flow is not removed. Error in kibana observed:
```
Flow 27Mar120649_874_walnut5615 error - RemoveRulesAction failed: Last flow(id:27Mar120649_874_walnut5615, path:27Mar120649_874_walnut5615_69f6d01d-03f9-4892-b6d0-0065dd43d38c) segment and flow level multi-table flags value are incompatible to each other - segment(false) != flow(true)
```

**Note:** Apart from directly fixing the delete procedure, we should recheck and asses whether given state of the flow(main path in multitable, protected in singletable) doesn't bring other related problems (flow update, flow reroute etc.)

**Comment by rtretyak** [comment]
Issue is still relevant as of now

**Comment by rtretyak** [comment]
Closed by #3946
